apps/io-wallet-user-func/src/email-provider.ts

```diff
@@ -0,0 +1,55 @@
+import emailTemplate from "@/templates/wallet-instance-activation/index.html";
+import {
+  getMailerTransporter,
+  sendMail,
+} from "@pagopa/io-functions-commons/dist/src/mailer";
+import * as RTE from "fp-ts/lib/ReaderTaskEither";
+import * as TE from "fp-ts/lib/TaskEither";
+import { pipe } from "fp-ts/lib/function";
+
+import { MailConfig } from "./app/config";
+
+export const sendEmail: (
+  fiscalCode: string,
+) => RTE.ReaderTaskEither<{ mail: MailConfig }, Error, void> =
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  (fiscalCode) => (configs) =>
+    pipe(
+      TE.tryCatch(
+        async () => {
+          if (!configs.mail.enabled) {
+            return;
+          }
+          const userEmail = ""; // to do - get the email by fiscalCode
+
+          await sendMail(
+            getMailerTransporter({
+              MAIL_FROM: configs.mail.mailhogHost?.length
+                ? configs.mail.mailSender
+                : userEmail,
+              MAIL_TRANSPORTS: undefined,
+              MAILHOG_HOSTNAME: configs.mail.mailhogHost?.length
+                ? configs.mail.mailhogHost
+                : undefined,
+              MAILUP_SECRET: configs.mail.mailupSecret?.length
+                ? configs.mail.mailupSecret
+                : undefined,
+              MAILUP_USERNAME: configs.mail.mailupUsername?.length
+                ? configs.mail.mailupUsername
+                : undefined,
+              NODE_ENV: "development",
```

Comment:
Why?

Reply:
(Same previous comment about the length check)

Reply:
sorry, the comment was referring to NODE_ENV.

Reply:
The `getMailerTransporter()` function is really pressing on the configurations. In practice it expects the variable `NODE_ENV` that for mailup must be valued exactly to the literal string `"production"`, otherwise the function itself throws an error similar to the following:

`Exception: Failed to choose a mail transport based on provided configuration`

Reply:
I think you could use `process.env.NODE_ENV`

Reply:
Oops, I don't think I can use `process.env.NODE_ENV` directly, unfortunately! This is because the type that expects getMailerTransporter includes this line of code in the type definition:

`NODE_ENV: t.LiteralC<"production">;`

I'll try to see if there's any other simple way!